commands/azureCommands/create-registry.ts

```diff
@@ -0,0 +1,160 @@
+    do {
+        subscriptionName = await vscode.window.showQuickPick(subsNames, { 'canPickMany': false, 'placeHolder': 'Choose a subscription to be used' });
+
+        if (subscriptionName === undefined) throw 'User exit';
+    } while (!subscriptionName);
+
+
+    return subs.find(sub => { return sub.displayName === subscriptionName });
+}
+
+async function acquireResourceGroup(subscription: SubscriptionModels.Subscription): Promise<ResourceGroup> {
+    //Acquire each subscription's data simultaneously
+    let resourceGroup;
+    let resourceGroupName;
+    const resourceGroupClient = new ResourceManagementClient(AzureCredentialsManager.getInstance().getCredentialByTenantId(subscription.tenantId), subscription.subscriptionId);
+    let resourceGroups = await AzureCredentialsManager.getInstance().getResourceGroups(subscription);
+
+    let resourceGroupNames: string[] = [];
+    resourceGroupNames.push('+ Create new resource group');
+    for (let i = 0; i < resourceGroups.length; i++) {
+        resourceGroupNames.push(resourceGroups[i].name);
+    }
+
+    do {
+        resourceGroupName = await vscode.window.showQuickPick(resourceGroupNames, { 'canPickMany': false, 'placeHolder': 'Choose a Resource Group to be used' });
+        if (resourceGroupName === undefined) throw 'user Exit';
+        if (resourceGroupName === '+ Create new resource group') {
+            let opt: vscode.InputBoxOptions = {
+                ignoreFocusOut: false,
+                prompt: 'Resource group name? '
+            };
+            resourceGroupName = await vscode.window.showInputBox(opt);
+            let resourceGroupStatus: boolean = await resourceGroupClient.resourceGroups.checkExistence(resourceGroupName);
+            while (resourceGroupStatus) {
+                opt = {
+                    ignoreFocusOut: false,
+                    prompt: "That resource group name is already in existence. Try again: "
+                }
+                resourceGroupName = await vscode.window.showInputBox(opt);
+                if (resourceGroupName === undefined) throw 'user Exit';
+                resourceGroupStatus = await resourceGroupClient.resourceGroups.checkExistence(resourceGroupName);
+            }
+
+            let newResourceGroup: ResourceGroup = {
+                name: resourceGroupName,
+                location: 'West US',
+            };
+            await resourceGroupClient.resourceGroups.createOrUpdate(resourceGroupName, newResourceGroup);
+        }
+
+        resourceGroups = await resourceGroupClient.resourceGroups.list();
+        resourceGroup = resourceGroups.find(resGroup => { return resGroup.name === resourceGroupName; });
+        if (!resourceGroupName) vscode.window.showErrorMessage('You must select a valid resource group');
+    } while (!resourceGroupName);
+
+    return resourceGroup;
+}
+
+async function acquireRegistryName(client: ContainerRegistryManagementClient) {
+    let opt: vscode.InputBoxOptions = {
```

Comment:
I see some dup codes here. Do you think you can just use the code in the while-loop to get and check the registry name? For example:
// pseudo code
let promptMessage = 'Registry name?'
let registryName = ''
while (promptMessage) {
  registryName = getRegistryName(promptMessage)
  if (checkRegistryName(registryName)) {
    promptMessage = null
  } else {
    promptMessage = "That registry name is unavailable. Try again: "
  }
}

return registryName;

Reply:
The code you highlighted is for new resource group name, not registry. Which prompt would you like me to change?

Reply:
I guess i mistake resource group name with registry name. The idea is using one while loop.

---
In reply to: [206290008](https://github.com/AzureCR/vscode-docker/pull/22#discussion_r206290008) [](ancestors = 206290008)

Reply:
Thanks, made that more efficient now